### ISSUE TITLE
ViewModel not applied 

### ISSUE THREAD

**Issue by kimlongkpop1** [googlesamples/android-architecture#640]
"Brand : todo-mvvm-live"

. when i rotate device , viewmodel.start has call again and reload data from repo ,the method replaceData is call twice because it load fist time from cache list in viewmodel and second time from list in repo, so the viewmodel architecture are not applied at this time , please fix this , tkanks

**Comment by JoseAlcerreca** [comment]
You're right, this is a legacy from presenters. You can just start loading in the VM's constructor.

**Comment by kimlongkpop1** [comment]
when observe TasViewModel.taskList in xml , the bindingAdapter method setItems is called even has no call to mTaskList.setValue(mList)  , after that the new call to setItems when have change in mTaskList , so i have 2 time  call to setItems binding method. When i observer it in Fragment , it only work when setValue is called , is this regular ? sorry for my bad english

**Comment by kimlongkpop1** [comment]
> You're right, this is a legacy from presenters. You can just start loading in the VM's constructor.

when observe TasViewModel.taskList in xml , the bindingAdapter method setItems is called even has no call to mTaskList.setValue(mList) , after that the new call to setItems when have change in mTaskList , so i have 2 time call to setItems binding method. When i observer it in Fragment , it only work when setValue is called , is this regular ? sorry for my bad english

**Comment by kimlongkpop1** [comment]
> You're right, this is a legacy from presenters. You can just start loading in the VM's constructor.

when observe TasViewModel.taskList in xml , the bindingAdapter method setItems is called  fist time with null taskList (but it is not null in TaskViewModel), after that the new call to setItems when have change in mTaskList , so i have 2 time call to setItems binding method. When i observer it in Fragment , it only work when setValue is called or taskList have change , is this regular ? sorry for my bad english

**Comment by jeffmcnd** [comment]
Would adding loading function calls in the constructor cause issues for testing? For example, we might have to provide mock classes that aren't relevant or spy the view model in order to test something else but methods are called in the constructor so we can't do anything about it.

Could we instead use the savedInstanceState variable to check to see if we should call loading methods? If the Bundle is null then we know to call load method(s) otherwise we don't need to call the load methods.

`override fun onCreate(savedInstanceState: Bundle?) {
    // ...
    if (savedInstanceState == null) {
        viewModel.load()
    }
`

**Comment by 33558903** [comment]
#640

**Comment by Yvajaem1y** [comment]
Sorry, but could you please clarify if the issue is still relevant? It's just that from what I can see in the logs, there doesn't seem to be a problem.
